public/app/plugins/datasource/cloudwatch/datasource.ts

```diff
@@ -66,10 +66,19 @@ export class CloudWatchDatasource
     this.languageProvider = new CloudWatchLogsLanguageProvider(this);
     this.sqlCompletionItemProvider = new SQLCompletionItemProvider(this.resources, this.templateSrv);
     this.metricMathCompletionItemProvider = new MetricMathCompletionItemProvider(this.resources, this.templateSrv);
-    this.metricsQueryRunner = new CloudWatchMetricsQueryRunner(instanceSettings, templateSrv);
```

Comment:
I wonder if there's a test we can add to ensure we always call super? Or if we could add a comment or something explaining what super.query inherits or something?

Reply:
lines 19-20 in `public/app/plugins/datasource/cloudwatch/__mocks__/CloudWatchDataSource.ts` should cover it i think. the tests in `public/app/plugins/datasource/cloudwatch/datasource.test.ts` are checking whether that queryMock is the one being called.